src/plots/cartesian/axes.js

```diff
@@ -611,31 +611,43 @@ axes.calcTicks = function calcTicks(ax) {
     generateTicks();
 
     if(ax.rangebreaks) {
-        var nTicksBefore = tickVals.length;
-
         // remove ticks falling inside rangebreaks
         tickVals = tickVals.filter(function(d) {
             return ax.maskBreaks(d.value) !== BADNUM;
         });
 
-        // if 'numerous' ticks get placed into rangebreaks,
-        // increase dtick to generate more ticks,
-        // so that some hopefully fall between rangebreaks
-        if(ax.tickmode === 'auto' && tickVals.length < nTicksBefore / 6) {
-            axes.autoTicks(ax, ax._roughDTick / 3);
-            autoTickRound(ax);
-            ax._tmin = axes.tickFirst(ax);
-            generateTicks();
-            tickVals = tickVals.filter(function(d) {
-                return ax.maskBreaks(d.value) !== BADNUM;
-            });
+        // add tick at the start of every rangebreak
```

Comment:
That's too much - we should only add ticks to breaks that would get a tick anyway.

Compare this screenshot from "before"
<img width="744" alt="Screen Shot 2020-03-23 at 6 09 03 PM" src="https://user-images.githubusercontent.com/2678795/77367889-73180300-6d31-11ea-8e97-6f87883dd958.png">

With a similar range "after"
<img width="717" alt="Screen Shot 2020-03-23 at 6 09 57 PM" src="https://user-images.githubusercontent.com/2678795/77367915-86c36980-6d31-11ea-9b38-fb6d45957c89.png">

That actually tried to make even more ticks than you see, but some later step removed lots of them because it couldn't draw them without overlapping.

I'm thinking the above `tickVals = tickVals.filter` needs to somehow turn into:
- Find all the ticks within the SAME break - note that this isn't necessarily "all the consecutive ticks with `maskBreaks === BADNUM`, because you could have two consecutive ticks in different breaks. For example, month ticks for February and March, in years when both of those start on a weekend (because February is exactly 4 weeks long). I don't know how to do that, we may need a new helper function that's effectively inverse of `maskBreaks`, so if you're in a break it returns the beginning (or end?) of that break.
- Collapse those all to a single tick, positioned at the break, but labeled with its original value.
- Previously I think we discussed using the first value, but now I'm thinking the value of the last tick would make more sense, particularly for date axes where currently the ticks essentially label the time period that follows them. For example, if you remove nights AND weekends but zoom to where ticks are daily, so that Saturday, Sunday, and Monday ticks are within the same break, you'd rather see the re-inserted tick say Monday rather than Saturday.

Reply:
@alexcjohnson thanks for the feedback.
I started some adjustments to use end of the break and reduce the effective length of axis by the total length of the breaks in  https://github.com/plotly/plotly.js/pull/4677/commits/7b627e53fbf10c25180a49df03dfb0ae3ef38029.
I am working on another commit to possibly improve this.